packages/schema/src/types/object.ts

```diff
@@ -377,7 +377,10 @@ function validateValueObject({
             ctxt.createChild(propTypePrefix + key, valueObject[key], schema)
           )
         );
-      } else if (schema.type().indexOf('Optional<') !== 0) {
+      } else if (
```

Comment:
lets create a function and move the condition
if (!isOptionalOrNullableType(schema.type())) {
}